src/app/query/bitcoin/blockstream-rate-limiter.ts

```diff
@@ -0,0 +1,23 @@
+import { RateLimiter } from 'limiter';
+
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+
+const blockstreamMainnetApiLimiter = new RateLimiter({
+  tokensPerInterval: 20,
+  interval: 5000,
+});
+
+const blockstreamTestnetApiLimiter = new RateLimiter({
+  tokensPerInterval: 20,
+  interval: 5000,
+});
+
+export function useBlockstreamRateLimiter() {
+  const currentNetwork = useCurrentNetworkState();
+
+  if (currentNetwork.isTestnet) {
+    return blockstreamTestnetApiLimiter;
+  }
+
+  return blockstreamMainnetApiLimiter;
+}
```

Comment:
This is the same approach I took for Hiro API, but actually wonder if it might be better to keep outside of React/hooks, and forward to correct limiter based on URL.

```ts
function getBlockstreamRatelimiter (url: string) {
  if (url.contains(checkForTestnetUrl)) return blockstreamTestnetApiLimiter;
  return blockstreamMainnetApiLimiter;
}
// await getBlockstreamRatelimiter(url).removeToken(1)
```

This would tidy things up a lot, as we don't need to pass the rate limiter to the fetcher fn, and remove dependence on React.

Reply:
e.g. this way of doing it

https://github.com/leather-wallet/extension/blob/dee8a8e41af3e9c2fdc0722402f3949cb8d1e668/src/app/query/stacks/balance/stx-balance.query.ts#L20-L46

could be way cleaner not passing the rate limiter around

Reply:
in most cases we know url in client api classes, like here:
```
async getUtxosByAddress(address: string, signal?: AbortSignal): Promise<UtxoResponseItem[]> {
    const resp = await axios.get<UtxoResponseItem[]>(
      `${this.configuration.baseUrl}/address/${address}/utxo`,
      {
        signal,
      }
    );
    return resp.data.sort((a, b) => a.vout - b.vout);
  }
```

you think better to put limiter there?🤔

Reply:
I'm not sure whether it's best to put it in here, or directly in the query, up to you.

But determining the correct limiter to use via the url vs react network state, is a better way to do it I think

Reply:
in most cases we don't know url on query level

Reply:
I see, yeah so we'd have to put the `RateLimiter` in the `getUtxosByAddress` directly. 

I don't think that's so bad, as rate limits are global rules we shouldn't ever need to circumvent?